### ISSUE TITLE
Lint warning doesn't appear until script is edited

### ISSUE THREAD

**Issue by the0neWhoKnocks** [sveltejs/eslint-plugin-svelte3#113]
Not sure if this is an Atom thing or an issue with this plugin... or both.

- Currently, if I add an unused rule within a `style` tag, no lint warnings appear.
- When I start to edit anything within a `script` tag, the lint warning appears.
- Same behavior to fix the warning. Remove bad code, go to `script` tag, add space or whatever, then the linter kicks in.

![lint-bug](https://user-images.githubusercontent.com/344140/119379294-2c46d580-bc74-11eb-8d85-a507dfca4cf4.gif)

**Comment by Conduitry** [comment]
This is almost certainly an issue with Atom. I'm guessing the ESLint integration has an optimization where it doesn't re-run linting unless the script tag is updated. I haven't used Atom in a few years, so if you find something in the configuration for their linter or linter-eslint packages that disables this optimization (or whatever needs to happen) a PR would be appreciated.

**Comment by the0neWhoKnocks** [comment]
I've got `linter-ui-default`, `linter-eslint`, and `linter` installed. The only one that has any settings that key off of file changes is `linter`. I've got everything enabled in `linter`
```
[x] Lint Preview Tabs
[x] Lint on Open
[x] Lint on Change
```
There is a note for `Lint on Change` that states:
>Lint files while typing, without the need to save (only for supported providers)

Not certain what `supported providers` refers to specifically though. Is it this plugin, the language server, or something else?

I'll debug `linter` and see if I can figure out the answer to what `supported providers` is.

**Comment by the0neWhoKnocks** [comment]
After some debugging, I found a comparison happening in `linter`
```js
function Ae(e, t, i) {
    return !(t && !e.lintsOnChange) && i.some((function(t) {
        return e.grammarScopes.includes(t)
    }))
}
```
`e.grammarScopes` always equaled `["source.js", "source.jsx", "source.js.jsx", "source.flow", "source.babel", "source.js-semantic"]`.

While editing within a `style` node, `i` equaled `["*", "source.svelte", "source.css", "meta.property-list.css"]`.

While editing within a `script` node, `i` equaled `["*", "source.svelte", "source.js"]`.

I wasn't able to determine where `i` was being defined, but `e.grammarScopes` was defined within `linter-eslint`. Once you click in that input, it fills it in with the defaults, I appended `source.svelte`, and am now getting realtime lint warnings.

![linter-eslint-svelte-grammar](https://user-images.githubusercontent.com/344140/119540076-d93b5400-bd41-11eb-808b-6e82a6612787.png)

Also, I used to have the `core: customFileTypes` stuff set up, but removed it while debugging and haven't noticed anything not working. Not sure if that doc is out of date, or required for another use case, but perhaps it can be removed.
